active/0116-qunit-dom.md

```diff
@@ -0,0 +1,129 @@
+- Start Date: 2018-02-12
+- RFC PR: #116
+
+# Summary
+
+Introduce [`qunit-dom`] as a dependency by default in the `app` and `addon` blueprints.
+
+[`qunit-dom`]: https://github.com/simplabs/qunit-dom
+
+# Motivation
+
+> Why are we doing this?
+
+In a modern Ember application making assertions around the state of the DOM is
+fundamental to confirming your applications functionality. These assertions are
+often quite verbose:
+
+```js
+assert.equal(this.element.querySelector('.title').textContent.trim(), 'Hello World!');
+```
+
+Using the `find()` helper of `@ember/test-helpers` we can simplify the DOM
+element lookup, but the signal-to-noise ratio of the code is still not great:
+
+```js
+assert.equal(find('.title').textContent.trim(), 'Hello World!');
+```
+
+With `qunit-dom` we can write much more readable assertions for DOM elements:
+
+```js
+assert.dom('.title').hasText('Hello World!');
+```
+
+> What use cases does it support?
+
+It supports the most common assertions on DOM elements, like:
+
+- what text does the element have?
+- what value does the `<input>` element have?
+- is a certain CSS class applied to the element
+
+The full API is documented at <https://github.com/simplabs/qunit-dom/blob/master/API.md>.
+
+> What is the expected outcome?
+
+Using `qunit-dom` will lead to more simple and readable test code.
+
+
+# Detailed design
+
+The necessary changes to `ember-cli` are relatively small since we only need
+to add the dependency to the `app` blueprint, and the `addon` blueprint will
+inherit it automatically.
+
+This has the advantage (over including it as an implicit dependency), that
+apps and addons that don't want to use it for some reason can opt-out by
+removing the dependency from their `package.json` file.
+
+A WIP pull request has been created already at <https://github.com/ember-cli/ember-cli/pull/7605>.
+
+
+# How We Teach This
+
+> Would the acceptance of this proposal mean the Ember guides must be
+> re-organized or altered? Does it change how Ember is taught to new users
+> at any level?
+
+Once we decide that this is the right way to go, we should update the official
+Ember.js testing guides to use `qunit-dom` assertions by default. This has the
```

Comment:
we should make sure to document the fact that app devs can opt out by removing the dependency from their `package.json`